lib/fpm/package.rb

```diff
@@ -249,8 +257,20 @@ def output(path)
                                   "creating #{self.type} packages")
   end # def output
 
+  def temporary_directory(path)
+    if attributes[:source_date_epoch].nil?
+      return Stud::Temporary.directory("package-#{type}-staging")
+    else
+      # Choose a predictable build path; see https://reproducible-builds.org/docs/build-path/
+      root = ENV["TMP"] || ENV["TMPDIR"] || ENV["TEMP"] || "/tmp"
```

Comment:
This seems somewhat regrettable (ie. it's not actually "predictable" as the comment claims IMHO) but I can see how there is no real workaround for fpm as it cannot, for example, guarantee the availability of, say, ``/tmp/buildd/whatever``.